homeassistant/components/drop_connect/sensor.py

```diff
@@ -0,0 +1,327 @@
+    ),
+    DROPSensorEntityDescription(
+        key=BATTERY,
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.drop_api.battery(),
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    DROPSensorEntityDescription(
+        key=TEMPERATURE,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        suggested_display_precision=1,
+        value_fn=lambda device: device.drop_api.temperature(),
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    DROPSensorEntityDescription(
+        key=INLET_TDS,
+        translation_key=INLET_TDS,
+        icon=TDS_ICON,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.drop_api.inlet_tds(),
+    ),
+    DROPSensorEntityDescription(
+        key=OUTLET_TDS,
+        translation_key=OUTLET_TDS,
+        icon=TDS_ICON,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.drop_api.outlet_tds(),
+    ),
+    DROPSensorEntityDescription(
+        key=CARTRIDGE_1_LIFE,
+        translation_key=CARTRIDGE_1_LIFE,
+        icon=GAUGE_ICON,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.drop_api.cart1(),
+    ),
+    DROPSensorEntityDescription(
+        key=CARTRIDGE_2_LIFE,
+        translation_key=CARTRIDGE_2_LIFE,
+        icon=GAUGE_ICON,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.drop_api.cart2(),
+    ),
+    DROPSensorEntityDescription(
+        key=CARTRIDGE_3_LIFE,
+        translation_key=CARTRIDGE_3_LIFE,
+        icon=GAUGE_ICON,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.drop_api.cart3(),
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the DROP sensors from config entry."""
+    _LOGGER.debug(
+        "Set up sensor for device type %s with entry_id is %s",
+        config_entry.data[CONF_DEVICE_TYPE],
+        config_entry.entry_id,
+    )
+
+    coordinator: DROPDeviceDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    device_type: str = config_entry.data[CONF_DEVICE_TYPE]
+    if device_type == DEV_HUB:
```

Comment:
May be a `device_type` to description key mapping is a bit cleaner?

Reply:
Changed in 75a9c6cc5cdf80ec927a899f92c82787814aa7ed.